numpy/core/tests/test_umath.py

```diff
@@ -789,6 +789,32 @@ def assert_complex_equal(x, y):
                 assert_complex_equal(np.power(zero, -p), cnan)
             assert_complex_equal(np.power(zero, -1+0.2j), cnan)
 
+    # Testing 0^{Non-zero} issue 18378
+    def test_zero_power_nonzero(self):
+        zero = np.array([0.0j])
```

Comment:
```suggestion
        zero = np.array([0.0+0.0j])
```

Reply:
@scratchmex I think this is better the way it was originally